### ISSUE TITLE
Failover happens differently with different pods

### ISSUE THREAD

**Issue by minac** [ucloud/redis-operator#18]
Hi! Thanks for the operator!

I would like to start using it, but came across a show stopper, in which the failover is not happening and the cluster is left leaderless for a while.

Here are the steps:
1. I created a cluster with 3 pods. The master has been allocated to the IP of pod `redis-cluster-test-2`. 
2. When I change a pod spec variable like `cpu requests`, it terminates `redis-cluster-test-2` and does an instant failover to the IP of `redis-cluster-test-0` or `redis-cluster-test-1`. So far so good.
3. Then one of two things occur:
  3A. If the master became the IP of `redis-cluster-test-1`, the sentinel service keeps pointing to that IP until both `redis-cluster-test-1` and `redis-cluster-test-0` have been recreated (about 80s)
  3B. If the master became the IP of `redis-cluster-test-0`, then `redis-cluster-test-1` gets recreated, and the sentinel keeps pointing to the IP of `redis-cluster-test-0`. When ``redis-cluster-test-0` gets recreated, the sentinel keeps pointing to the old IP until the recreation is complete and a few seconds more (about 40s).

During this time the Redis cluster is leaderless and any writes fail because the sentinels are still pointing to an IP that is no longer in use.

In other words, the failover is only being forced on the first pod to be terminated, which, in a statefulset is pod `-2`.

Can you please help @polefishu ?

Thanks!

**Comment by minac** [comment]
I think I found out what it was, there is a `bash`ism in the `sh` script and the sentinels don't always reply in the custom shutdown script. 

I commented the `pipefail` instruction which works in `bash` but not `sh`, and adjusted a couple of others.
Then I added a while loop around `redis-cli` commands to ensure we get a proper response:

```
#!/usr/bin/env sh
#set -eou pipefail
master=""
response_code=""
while [ "$master" = "" ]; do
	echo "Asking sentinel who is master..."
	master=$(redis-cli -h ${REDIS_SENTINEL_TEST_SERVICE_HOST} -p ${REDIS_SENTINEL_TEST_SERVICE_PORT_SENTINEL} --csv SENTINEL get-master-addr-by-name mymaster | tr ',' ' ' | tr -d '\"' |cut -d' ' -f1)
	sleep 3
done
echo "Master is $master, doing redis save..."
redis-cli SAVE
if [ $master = $(hostname -i) ]; then
	while [ ! "$response_code" = "OK" ]; do
	  response_code=$(redis-cli -h ${REDIS_SENTINEL_TEST_SERVICE_HOST} -p ${REDIS_SENTINEL_TEST_SERVICE_PORT_SENTINEL} SENTINEL failover mymaster)
	  sleep 3
	  echo "after failover with code $response_code"
	done
fi
```

I hope this helps.

**Comment by polefishu** [comment]
Hi @minac, Thanks for your interest on the redis-operator!
How did you modify the `cpu requests` ?
Directly modify the `statefulset.spec.template.spec` or the `redisclusters.spec.resources`?

**Comment by minac** [comment]
Hi @polefishu!
I modified it in the `redisclusters.spec.resources`.

**Comment by polefishu** [comment]
Thinks, I will fix this issue in the new version.
